### PR TITLE
Generate two API docs (full/public API). Add public API check.

### PR DESCRIPTION
**Pull request by qinsoon**
With this pull request, we will generate two API docs (one for full MMTk API including private items `--document-private-items`, and one for public API). They will be hosted in:
* https://www.mmtk.io/mmtk-core/public-doc/
* https://www.mmtk.io/mmtk-core/full-doc/

This pull request also adds a public API check for pull requests. If a pull request changes the public API, the action will fail. Ideally the action returns a neutral status. However, that is not supported by Github action yet.

### CODE REVIEW

**Comment by qinsoon** [comment]
The document link in https://www.mmtk.io/code needs to be updated once we merge this PR.

File: README.md [hunk 1]
```diff
@@ -46,7 +46,9 @@ The artefact produced produced by the build process can be found under
 MMTk does not run standalone. You would need to integrate MMTk with a language implementation.
 You can either try out one of the VM bindings we have been working on, or implement your own binding in your VM for MMTk.
 You can also implement your own GC algorithm in MMTk, and run it with supported VMs.
-You can find an up-to-date API document for mmtk-core here: https://www.mmtk.io/mmtk-core/mmtk.
+You can find an up-to-date API document for mmtk-core here:
+* If you are trying to port MMTk to your language, check our public API: https://www.mmtk.io/mmtk-core/public-api
+* If you are trying to develop in MMTk (e.g. a new GC algorithm), check our full API: https://www.mmtk.io/mmtk-core/full-api
```
Comment (wks) [inline README.md:51]: I prefer not to call the full documentation an "API".  API stands for "application programming interface", and it is for those who develop applications (in our case, VMs).  I suggest "full documentation" and "full-doc".

Comment (qinsoon) [reply README.md]: I have made some changes based on your suggestion. Can you check again?

**Review by wks — APPROVED** [review]
LGTM.  Please make sure the web pages are properly deployed, and the hyperlinks from https://www.mmtk.io/ are correct, too.

**Pull request merged**